### PR TITLE
Various Channel improvements

### PR DESCRIPTION
**Pull request by pakrym**
Add `ReadableBuffer.TrimEnd`
Add location we are consuming from to `Already consuming.` error message.
Add `ReadBufferAsync` extension method that returns buffer immediately.
Add `ReadableBufferCreate(byte[])`

### CODE REVIEW

**Comment by davidfowl** [comment]
This PR needs some tests

File: src/Channels/ChannelExtensions.cs [hunk 1]
```diff
@@ -68,6 +68,11 @@ public static ValueTask<int> ReadAsync(this IReadableChannel input, Span<byte> d
             return new ValueTask<int>(input.ReadAsyncAwaited(destination));
         }
 
+        public static async Task<ReadableBuffer> ReadBufferAsync(this IReadableChannel input)
```
Comment (davidfowl) [inline src/Channels/ChannelExtensions.cs]: Is this only for testing?

File: src/Channels/MemoryEnumerator.cs [hunk 2]
```diff
@@ -19,7 +17,7 @@ public struct MemoryEnumerator
         /// <summary>
         /// 
         /// </summary>
-        public MemoryEnumerator(ref ReadCursor start, ref ReadCursor end)
+        public MemoryEnumerator(ReadCursor start, ReadCursor end)
```
Comment (davidfowl) [inline src/Channels/MemoryEnumerator.cs:20]: 👍

File: src/Channels/Channel.cs [hunk 3]
```diff
@@ -318,6 +320,7 @@ private void Complete()
                 !ReferenceEquals(awaitableState, _awaitableIsNotCompleted))
             {
                 awaitableState();
+                // Task.Run(awaitableState);
```
Comment (davidfowl) [inline src/Channels/Channel.cs]: Remove

**Review by davidfowl — CHANGES_REQUESTED** [review]

File: src/Channels/ChannelExtensions.cs (hunk 1 above)
Comment (pakrym) [reply src/Channels/ChannelExtensions.cs]: I'm not sure. I use it only in testing.

File: src/Channels/Channel.cs (hunk 3 above)
Comment (pakrym) [reply src/Channels/Channel.cs]: Done

File: src/Channels/ChannelExtensions.cs (hunk 1 above)
Comment (davidfowl) [reply src/Channels/ChannelExtensions.cs]: Move it there

Comment (pakrym) [reply src/Channels/ChannelExtensions.cs]: Done

**Review by davidfowl — APPROVED** [review]

**Pull request merged**